manifests/init.pp

```diff
@@ -1,116 +1,52 @@
-#   sleep more, bigger sleep less.  How many Milliseconds to sleep between
-#   scans of 16GB of RAM.  The actual sleep time is calculated as sleep =
-#   KSM_SLEEP_MSEC * 16 / Total GB of RAM The final sleep value will be
-#   written to /sys/kernel/mm/ksm/sleep_millisecs
+# @param load_kernel_modules
+#   Manage kernel modules from this module
 #
-# @param ksm_npages_boost
-#   Amount to increment the number of pages to scan. The number of pages to
-#   be scanned will be increased by KSM_NPAGES_BOOST when the amount of free
-#   ram < threshold (see KSM_THRES_* below)
+# @param manage_sysctl
+#   Manage associated sysctl settings from this module
 #
-# @param ksm_npages_decay
-#   Amount to decrease the number of pages to scan The number of pages to be
-#   scanned will be decreased by KSM_NPAGES_DECAY when the amount of free ram
-#   >= threshold (see KSM_THRES_* below)
-#
-# @param ksm_npages_min
-#   Minimum number of pages to be scanned at all times
-#   If this variable is set to 'shmall', then half of the value in
-#   /proc/sys/kernel/shmall will be used.
-#
-# @param ksm_npages_max
-#   Maximum number of pages to be scanned at all times
-#   If this variable is set to 'shmall', then the value in
-#   /proc/sys/kernel/shmall will be used.
-#
-# @param ksm_thres_coef
-#   If free memory is less than this percentage KSM will be activated.
-#   NOTE: Only KSM_THRES_CONST or KSM_THRES_COEF is actually used.  Whichever
-#   results in a larger number wins.
-#
-# @param ksm_thres_const
-#   If free memory is less than this number KSM will be activated
-#   NOTE: Only KSM_THRES_CONST or KSM_THRES_COEF is actually used.  Whichever
-#   results in a larger number wins.
-#
-# @param kvm_package_list List of packages related to kvm to be managed
-# @param manage_sysctl Disable to avoid setting sysctl settings in this module.
-# @param package_ensure Ensure setting for all packages in this module
+# @param package_ensure
+#   Ensure setting for all packages in this module
 #
 # @author https://github.com/simp/pupmod-simp-libvirt/graphs/contributors
 #
 class libvirt (
-  Array[String] $package_list,
-  String $service_ensure,
-  Boolean $ksm,
-  Boolean $kvm,
-  Boolean $load_kernel_modules,
-
-  # KSM
-  Optional[Integer]               $ksm_max_kernel_pages,
-  Integer                         $ksm_monitor_interval,
-  Integer                         $ksm_sleep_msec,
-  Integer                         $ksm_npages_boost,
-  Integer[default,0]              $ksm_npages_decay,
-  Variant[Enum['shmall'],Integer] $ksm_npages_min,
-  Variant[Enum['shmall'],Integer] $ksm_npages_max,
-  Integer                         $ksm_thres_coef,
-  Optional[Integer]               $ksm_thres_const,
-
-  # KVM
-  Array[String] $kvm_package_list,
-  Boolean $manage_sysctl,
-
-  String $package_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
+  Array[String]  $package_list        = [
```

Comment:
this gets merged with the values from data? if it does, i think all default values should be in data

Reply:
@jeefberkey If you do that, `strings` has nothing useful to output. This is a style issue that we need to nail down for the infrastructure. Right now, this is consistent with the remainder of the modules and there is **no** data in the `data` directory for default values so that `strings` is valuable.